### ISSUE TITLE
IndexError: list index out of range - AT runid = commitMessageArray[2] 

### ISSUE THREAD

**Issue by joshuadmatthews** [Azure/gitops-connector#37]
I was not seeing the dispatch event on deploy completion, so went to the logs and noticed there is an index out of range error where it tries to fetch the runid/commitId. 

```
    def _get_source_commit_id_run_id(self, manifest_commitid):
        commitMessage = self.git_repository.get_commit_message(manifest_commitid)
        commitMessageArray = commitMessage.split('/', 5)
        runid = commitMessageArray[2]
        commitid = commitMessageArray[3]
        logging.info(f'CommitId {commitid}')
        return commitid, runid
```
Looks like it is expecting a commit message to exist on the manifest update commit with these values. I haven't seen that documented anywhere though, and in my case I am using flux image automation to automatically update manifests when a new image hits the container registry, so I don't think I will have any easy way to generate this commit message from flux.

**Comment by joshuadmatthews** [comment]
Also, I think assuming that there is only a single runId/commitId behind a manifest commit is not a good assumption. In my workflows there are dozens of different services owned by different teams each with their own workflows to publish containers. If multiple services updates happen at the same time, they may be picked up as a group by flux image automation controller and the manifests are updated in a single commit. 

In order to make a gitops connector flow work, I need to pass an array of runIds and commitId that represent all of the container images updated in this manifest commit. Then the publish workflow would then use these arrays to package up all of the necessary services.

**Comment by eedorenko** [comment]
Yes, gitops connector expects a commit message to exist in the manifests commit. Agree, it should be documented explicitly. GitOps connector "dispatch event" functionality is supposed to work in the flows similar to [this one](https://github.com/Azure/arc-cicd-demo-src/blob/master/docs/azdo-gitops-githubfluxv2.md) - [tutorial](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/tutorial-gitops-flux2-ci-cd#implement-cicd-with-github).

**Comment by joshuadmatthews** [comment]
I think it would be more flexible with error handling in case the message isn’t in the expected format,

also would be great if there was a third parameter sent that contains the whole commit message without parsing, this would allow user to pass whatever data is necessary through the commit message.

**Comment by eedorenko** [comment]
Agree! PRs are welcome :)

**Comment by joshuadmatthews** [comment]
Sure will give it a try. I’m not too familiar with python but seems simple enough.

**Comment by eedorenko** [comment]
Fixed in #39

**Comment by mr-onion-2** [comment]
> Yes, gitops connector expects a commit message to exist in the manifests commit. Agree, it should be documented explicitly. GitOps connector "dispatch event" functionality is supposed to work in the flows similar to [this one](https://github.com/Azure/arc-cicd-demo-src/blob/master/docs/azdo-gitops-githubfluxv2.md) - [tutorial](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/tutorial-gitops-flux2-ci-cd#implement-cicd-with-github).


Hi

Hope it's OK to comment on a closed issue, but I am curious what the format of the commit message is actually _supposed_ to be. The commit message generated in the example repo create-pr.sh script is `deployment $DEPLOY_ID` - and not a '/' delimited string containing the runId and commitId, which appears to be expected by the _get_source_commit_id_run_id_commit_mesage function ? 

I do have it working but only by making some tweak to the  function to work with my commit messages, e.g.  `Merge pull request #29 from <org>/<cluster>\n\n<commit message> | <runId>`
